cpu/sam0_common/sam0_eth/eth-netdev.c

```diff
@@ -44,26 +53,124 @@ extern bool sam0_eth_has_queued_pkt(void);
 extern void sam0_eth_set_mac(const eui48_t *mac);
 extern void sam0_eth_get_mac(char *out);
 extern void sam0_clear_rx_buffers(void);
+extern unsigned sam0_read_phy(uint8_t phy, uint8_t addr);
+extern void sam0_write_phy(uint8_t phy, uint8_t addr, uint16_t data);
+static void _restart_an(void *ctx);
 
 /* SAM0 CPUs only have one GMAC IP, so it is safe to
 statically defines one in this file */
 static sam0_eth_netdev_t _sam0_eth_dev;
 
+/* auto-negotiation timeout timer */
+static ztimer_t _phy_tim = { .callback = _restart_an };
+/* PHY interrupt status register */
+static uint16_t _phy_irq;
+
+static inline bool _get_link_status(void)
+{
+    return sam0_read_phy(0, MII_BMSR) & MII_BMSR_LINK;
```

Comment:
```suggestion
    return !!sam0_read_phy(0, MII_BMSR) & MII_BMSR_LINK;
```
or
```suggestion
    return sam0_read_phy(0, MII_BMSR) && MII_BMSR_LINK;
```

Since we want to return a boolean.